src/test/java/reposense/model/RepoLocationTest.java

```diff
@@ -1,49 +1,116 @@
-        assertInvalidLocation("https://github.com/reposense/RepoSense-git");
-        assertInvalidLocation("https://github.com/reposense/RepoSense.gi");
-        assertInvalidLocation("https://github.com/reposense/RepoSense");
+    public void isLocalRepo_validLocalRepos_success() throws Exception {
+        Assert.assertTrue(isLocalRepo(LOCAL_REPO_LOCATION_VALID_WITH_DOT_GIT_ONE));
+        Assert.assertTrue(isLocalRepo(LOCAL_REPO_LOCATION_WINDOWS_VALID_MIXED));
+        Assert.assertTrue(isLocalRepo("./abc:def"));
     }
 
     @Test
-    public void repoLocation_repoUrlWithASpecifiedBranch_throwsInvalidLocationException() {
-        // reject both repos with and without .git
-        assertInvalidLocation("https://github.com/reposense/testrepo-Beta/tree/add-config-json");
-        assertInvalidLocation("https://github.com/reposense/testrepo-Beta.git/tree/add-config-json");
+    public void isLocalRepo_remoteRepos_returnsFalse() throws Exception {
+        Assert.assertFalse(isLocalRepo("https://github.com/reposense/RepoSense.git/"));
+        Assert.assertFalse(isLocalRepo("git@github.com:reposense/RepoSense.git/"));
     }
 
     @Test
-    public void repoLocationParser_parseEmptyString_success() throws Exception {
-        RepoLocation repoLocation = new RepoLocation("");
+    public void repoLocation_parseLocalRepoLocation_success() throws Exception {
+        // local paths not containing ".git" should be valid
+        assertValidLocation(LOCAL_REPO_LOCATION_VALID_WITHOUT_DOT_GIT_ONE,
+                EXPECTED_REPO_NAME, EXPECTED_ORGANIZATION);
+        // relative pathing should be considered part of the 'organization' for differentiation
+        assertValidLocation(LOCAL_REPO_LOCATION_VALID_WITHOUT_DOT_GIT_TWO,
+                EXPECTED_REPO_NAME, "..-" + EXPECTED_ORGANIZATION);
+
+        // local paths containing ".git" should also be valid
+        assertValidLocation(LOCAL_REPO_LOCATION_VALID_WITH_DOT_GIT_ONE,
+                EXPECTED_REPO_NAME, EXPECTED_ORGANIZATION);
+    }
+
+    @Test
+    public void repoLocation_parseWindowsLocalRepoLocation_success() throws Exception {
+        Assume.assumeTrue(SystemUtil.isWindows());
+        // repeated tests but with windows file separators
+        assertValidLocation(LOCAL_REPO_LOCATION_WINDOWS_VALID_WITHOUT_DOT_GIT_ONE,
+                EXPECTED_REPO_NAME, EXPECTED_ORGANIZATION);
+        assertValidLocation(LOCAL_REPO_LOCATION_WINDOWS_VALID_WITHOUT_DOT_GIT_TWO,
+                EXPECTED_REPO_NAME, "..-" + EXPECTED_ORGANIZATION);
+        assertValidLocation(LOCAL_REPO_LOCATION_WINDOWS_VALID_WITH_DOT_GIT_ONE,
+                EXPECTED_REPO_NAME, EXPECTED_ORGANIZATION);
+
+        assertValidLocation(LOCAL_REPO_LOCATION_WINDOWS_VALID_MIXED,
+                EXPECTED_REPO_NAME, "..-" + EXPECTED_ORGANIZATION);
+    }
+
+    @Test
+    public void repoLocation_parseValidRemoteRepoUrl_success() throws Exception {
+        // valid url without specifying branch
```

Comment:
Let's add BitBucket as well to support the top 3 websites, the rest I think we can just ignore

Reply:
Noted. Do we have a bitbucket reposense remote repo? Else I will add in a random bitbucket repository.

Reply:
Nope, we do not. I think the URL you used is fine.